pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/BuildWheelsTask.groovy

```diff
@@ -147,105 +137,22 @@ class BuildWheelsTask extends DefaultTask implements SupportsWheelCache, Support
-                if (wheel.isPresent()) {
-                    File wheelFile = wheel.get()
-                    FileUtils.copyFile(wheelFile, new File(wheelExtension.wheelCache, wheelFile.name))
-                    if (PythonHelpers.isPlainOrVerbose(project)) {
-                        LOGGER.lifecycle("Skipping {}, in wheel cache {}", shortHand, wheelFile)
-                    }
-                    return
-                }
-
-                def tree = project.fileTree(
-                    dir: wheelExtension.wheelCache,
-                    include: "**/${ packageInfo.name.replace('-', '_') }-${ (packageInfo.version ?: 'unspecified').replace('-', '_') }-*.whl")
-
-                if (tree.files.size() >= 1) {
-                    LOGGER.lifecycle("Skipping {} wheel - Installed", shortHand)
-                    return
-                }
-            }
-
-            if (PythonHelpers.isPlainOrVerbose(project)) {
-                LOGGER.lifecycle("Installing {} wheel", shortHand)
-            }
-
-            def stream = new ByteArrayOutputStream()
-
-            def mergedEnv = environmentMerger.mergeEnvironments(
-                [pythonExtension.pythonEnvironment, environment, packageSettings.getEnvironment(packageInfo)])
-
-            def commandLine = [
-                pythonDetails.getVirtualEnvInterpreter().toString(),
-                pythonDetails.getVirtualEnvironment().getPip().toString(),
-                'wheel',
-                '--disable-pip-version-check',
-                '--wheel-dir', wheelExtension.wheelCache.toString(),
-                '--no-deps',
-            ]
-
-            commandLine.addAll(args)
-
-            def globalOptions = packageSettings.getGlobalOptions(packageInfo)
-            if (globalOptions != null) {
-                commandLine.addAll(globalOptions)
-            }
-
-            def buildOptions = packageSettings.getBuildOptions(packageInfo)
-            if (buildOptions != null) {
-                commandLine.addAll(buildOptions)
-            }
-
-            commandLine.add(installable.toString())
-
-            ExecResult installResult = project.exec { ExecSpec execSpec ->
-                execSpec.environment mergedEnv
-                execSpec.commandLine(commandLine)
-                execSpec.standardOutput = stream
-                execSpec.errorOutput = stream
-                execSpec.ignoreExitValue = true
-            }
-
-            if (installResult.exitValue != 0) {
-                LOGGER.error("Error installing package using `{}`", commandLine)
-                LOGGER.error(stream.toString().trim())
-                throw new GradleException("Failed to build wheel for ${ shortHand }. Please see above output for reason, or re-run your build using ``--info`` for additional logging.")
-            } else {
-                LOGGER.info(stream.toString().trim())
-            }
+            wheelAction.buildWheel(packageInfo, args)
```

Comment:
The `PipInstallTask` has a check for `packageExcludeFilter` here in case the dependency is conditional for a certain Python version and needs to be excluded.

We do not do that check neither here, nor in the `wheelAction.buildWheel()` method.
That needs to be fixed to have the code that is equivalent to the old one.

Reply:
Fixed. I missed that.